docs/modules/ROOT/pages/api/erc721.adoc

```diff
@@ -0,0 +1,648 @@
+
+See <<IERC721-get_approved,IERC721::get_approved>>.
+
+[.contract-item]
+[[ERC721-isApprovedForAll]]
+==== `[.contract-item-name]#++isApprovedForAll++#++(self: @ContractState, owner: ContractAddress, operator: ContractAddress) -> bool++` [.item-kind]#external#
+
+See <<IERC721-is_approved_for_all,IERC721::is_approved_for_all>>.
+
+[.contract-item]
+[[ERC721-tokenURI]]
+==== `[.contract-item-name]#++tokenURI++#++(self: @ContractState, tokenId: u256) -> felt252++` [.item-kind]#external#
+
+See <<IERC721Metadata-token_uri,IERC721Metadata::token_uri>>.
+
+==== Internal functions
+
+[.contract-item]
+[[ERC721-initializer]]
+==== `[.contract-item-name]#++initializer++#++(ref self: ContractState, name_: felt252, symbol_: felt252)++` [.item-kind]#internal#
+
+Initializes the contract by setting the token name and symbol.
+This should be used inside the contract's constructor.
+
+[.contract-item]
+[[ERC721-_owner_of]]
+==== `[.contract-item-name]#++_owner_of++#++(self: @ContractState, token_id: felt252) -> ContractAddress++` [.item-kind]#internal#
+
+Internal function that returns the owner address of `token_id`.
+This function will panic if the token does not exist.
+
+[.contract-item]
+[[ERC721-_exists]]
+==== `[.contract-item-name]#++_exists++#++(self: @ContractState, token_id: u256) -> bool++` [.item-kind]#internal#
+
+Internal function that returns whether `token_id` exists.
+
+Tokens start existing when they are minted (<<ERC721-_mint,_mint>>), and stop existing when they are burned (<<ERC721-_burn,_burn>>).
+
+[.contract-item]
+[[ERC721-_is_approved_or_owner]]
+==== `[.contract-item-name]#++_is_approved_or_owner++#++(ref self: ContractState, spender: ContractAddress, token_id: u256) -> bool++` [.item-kind]#internal#
+
+Internal function that returns whether `spender` is allowed to manage `token_id`.
+
+Requirements:
+
+- `token_id` exists.
+
+[.contract-item]
+[[ERC721-_approve]]
+==== `[.contract-item-name]#++_approve++#++(ref self: ContractState, to: ContractAddress, token_id: u256)++` [.item-kind]#internal#
+
+Internal function that changes or reaffirms the approved address for an NFT.
+
+Emits an <<IERC721-Approval,Approval>> event.
+
+Requirements:
+
+- `token_id` exists.
+- `to` is not the current token owner.
+
+[.contract-item]
+[[ERC721-_set_approval_for_all]]
+==== `[.contract-item-name]#++_set_approval_for_all++#++(ref self: ContractState, owner: ContractAddress, operator: ContractAddress, approved: bool)++` [.item-kind]#internal#
+
+Internal function that enables or disables approval for `operator` to manage all of the
+`owner` assets.
+
+Emits an <<IERC721-Approval,Approval>> event.
+
+Requirements:
+
+- `operator` cannot be the caller.
+
+[.contract-item]
+[[ERC721-_mint]]
+==== `[.contract-item-name]#++_mint++#++(ref self: ContractState, to: ContractAddress, token_id: u256)++` [.item-kind]#internal#
+
+WARNING: Usage of this method is discouraged, use <<IERC721-_safe_mint,_safe_mint>> whenever possible.
```

Comment:
same as in the ERC721 PR, why is this discouraged? they're different with their tradeoffs, imo none is inherently more secure than the other

Reply:
Valid point, will update